src/main/config/config_eeprom.c

```diff
@@ -254,34 +236,33 @@ static bool writeSettingsToEEPROM(void)
             // write the only instance
             record.flags |= CR_CLASSICATION_SYSTEM;
             config_streamer_write(&streamer, (uint8_t *)&record, sizeof(record));
-            chk = updateChecksum(chk, (uint8_t *)&record, sizeof(record));
+            crc = crc16_ccitt_update(crc, (uint8_t *)&record, sizeof(record));
             config_streamer_write(&streamer, reg->address, regSize);
-            chk = updateChecksum(chk, reg->address, regSize);
+            crc = crc16_ccitt_update(crc, reg->address, regSize);
         } else {
             // write one instance for each profile
             for (uint8_t profileIndex = 0; profileIndex < MAX_PROFILE_COUNT; profileIndex++) {
                 record.flags = 0;
 
                 record.flags |= ((profileIndex + 1) & CR_CLASSIFICATION_MASK);
                 config_streamer_write(&streamer, (uint8_t *)&record, sizeof(record));
-                chk = updateChecksum(chk, (uint8_t *)&record, sizeof(record));
+                crc = crc16_ccitt_update(crc, (uint8_t *)&record, sizeof(record));
                 const uint8_t *address = reg->address + (regSize * profileIndex);
                 config_streamer_write(&streamer, address, regSize);
-                chk = updateChecksum(chk, address, regSize);
+                crc = crc16_ccitt_update(crc, address, regSize);
             }
         }
     }
 
     configFooter_t footer = {
-        .terminator = 0,
```

Comment:
IMO terminator is used to mark end of EEPROM record, it is in place of `if (record->size == 0) {`